File: MyWorld/MyWorld/Controllers/API/TripsController.cs
using AutoMapper;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using MyWorld.Models;
using MyWorld.ViewModels;
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace MyWorld.Controllers.API
{
    [Route("api/trips")]
    [Authorize]
    public class TripsController : Controller
    {
        private IWorldRepository _repo;
        private ILogger<TripsController> _logger;

        public TripsController(IWorldRepository repo, ILogger<TripsController> logger)
        {
            _repo = repo;
            _logger = logger;
        }

        [HttpGet]
        public IActionResult Get()
        {
            try
            {
                IEnumerable<Trip> results = _repo.GetAllUserTrips(User.Identity.Name);
                return Ok(Mapper.Map<IEnumerable<TripViewModel>>(results));
            }
            catch (Exception ex)
            {
                _logger.LogError($"-- Exception in Get() of TripsController: {ex}");
                return BadRequest("> Error Occured.");
            }
        }

        [HttpPost]
        public async Task<IActionResult> Post([FromBody] TripViewModel theTrip)
        {
            if (ModelState.IsValid)
            {
                var newtrip = Mapper.Map<Trip>(theTrip);
                newtrip.UserName = User.Identity.Name;
                _repo.AddTrip(newtrip);

                if (await _repo.SaveChangesAsync())
                {
                    return Created($"api/trips/{theTrip.Name}", Mapper.Map<TripViewModel>(newtrip));
                }
            }
            return BadRequest("Failed to Save the trip");
        }
    }
}

File: MyWorld/MyWorld/wwwroot/js/tripsController.js
﻿// tripsController.js
(function () {
    "use strict";

    // Getting the existing module
    angular.module("app-trips")
        .controller("tripsController", tripsController);

    function tripsController($http) {
        var vm = this;

        vm.trips = [];

        vm.newTrip = {};

        vm.errorMessage = "";
        vm.isBusy = true;

        $http.get("/api/trips")
            .then(function (response) {
                // Success
                angular.copy(response.data, vm.trips);
            }, function (error) {
                // Failure
                vm.errorMessage = "Failed to load data: " + error;
            })
            .finally(function () {
                vm.isBusy = false;
            });

        // Add a new Trip
        vm.addTrip = function () {
            vm.errorMessage = "";
            vm.isBusy = true;

            $http.post("/api/trips", vm.newTrip)
                .then(function (response) {
                    vm.trips.push(response.data);
                    vm.newTrip = {};
                }, function (error) {
                    vm.errorMessage = "Failed to save new trip: " + error;
                })
                .finally(function () {
                    vm.isBusy = false;
                });
        }
    }
})();